mmap in podman container returns ENODEV on MacOS Host

### Issue Description

I'm trying to use cache files in a podman container. The cache files are shared across multiple `podman run` command. I used `mmap` to map the cache file into memory. That works fine on Linux host, but fails on MacOS host, with `ENODEV` error.

### Steps to reproduce the issue

Consider the following Dockerfile and script. This script is supposed to compile and run a simple C program that uses mmap to map a file into memory. The file is mounted into the container using a volume mount.

1. Create a directory and put the Dockerfile and script in it.
`Dockerfile`
```dockerfile
FROM fedora:latest

RUN dnf -y install gcc \
    && dnf clean all
```

`run.sh`
```bash
#!/bin/bash

set -xeu

podman build -t test .

MOUNT_DIR=$(pwd)/mountdir

mkdir -p ${MOUNT_DIR}

cat <<EOF > ${MOUNT_DIR}/test.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/mman.h>

int main(int argc, char *argv[]) {
    if (argc != 2) {
        fprintf(stderr, "Usage: %s <file>\n", argv[0]);
        exit(EXIT_FAILURE);
    }

    const char* input_file = argv[1];

    int result = 0;
    int fd;
    size_t size;
    void *addr;

    FILE* fp = fopen(input_file, "r");
    if (fp == NULL) {
        perror("fopen");
        result = EXIT_FAILURE;
        goto cleanup;
    }

    fd = fileno(fp);
    if (fd == -1) {
        perror("fileno");
        result = EXIT_FAILURE;
        goto cleanup;
    }

    if (fseek(fp, 0, SEEK_END) == -1) {
        perror("fseek");
        result = EXIT_FAILURE;
        goto cleanup;
    }

    size = ftell(fp);
    if (size == -1) {
        perror("ftell");
        result = EXIT_FAILURE;
        goto cleanup;
    }

    addr = mmap(NULL, size, PROT_READ, MAP_SHARED, fd, 0);
    if (addr == MAP_FAILED) {
        perror("mmap");
        result = EXIT_FAILURE;
        goto cleanup;
    }

    printf("size: %zu\n", size);
    printf("addr: %p\n", addr);

    if (munmap(addr, size) == -1) {
        perror("munmap");
        result = EXIT_FAILURE;
    } else {
        result = EXIT_SUCCESS;
    }

cleanup:
  if (fp != NULL) {
    fclose(fp);
  }

  return result;
}
EOF

podman run --rm -v ${MOUNT_DIR}:/mnt test bash -c "gcc -o /mnt/a.out /mnt/test.c && /mnt/a.out /mnt/test.c"
```

2. Run the `run.sh` script

### Describe the results you received

The script works fine on Linux, but fails on MacOS with the following error:
```
mmap: No such device
```
which means `ENODEV`.

### Describe the results you expected

Returns a valid address and size. For example, on Linux, the output is:
```
size: 1313
addr: 0x7f8a3333c000
```

### podman info output

```yaml
$ podman version
Client:       Podman Engine
Version:      4.6.1
API Version:  4.6.1
Go Version:   go1.20.7
Git Commit:   f3069b3ff48e30373c33b3f5976f15abf8cfee20
Built:        Fri Aug 11 03:13:43 2023
OS/Arch:      darwin/arm64

Server:       Podman Engine
Version:      4.6.0
API Version:  4.6.0
Go Version:   go1.20.6
Built:        Fri Jul 21 21:22:41 2023
OS/Arch:      linux/arm64
$ podman info
host:
  arch: arm64
  buildahVersion: 1.31.0
  cgroupControllers:
  - cpuset
  - cpu
  - io
  - memory
  - pids
  - rdma
  - misc
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.7-2.fc38.aarch64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.7, commit: '
  cpuUtilization:
    idlePercent: 96.64
    systemPercent: 2.94
    userPercent: 0.42
  cpus: 10
  databaseBackend: boltdb
  distribution:
    distribution: fedora
    variant: coreos
    version: "38"
  eventLogger: journald
  freeLocks: 2048
  hostname: localhost.localdomain
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 6.4.7-200.fc38.aarch64
  linkmode: dynamic
  logDriver: journald
  memFree: 7827574784
  memTotal: 8299237376
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    dns:
      package: aardvark-dns-1.7.0-1.fc38.aarch64
      path: /usr/libexec/podman/aardvark-dns
      version: aardvark-dns 1.7.0
    package: netavark-1.7.0-1.fc38.aarch64
    path: /usr/libexec/podman/netavark
    version: netavark 1.7.0
  ociRuntime:
    name: crun
    package: crun-1.8.6-1.fc38.aarch64
    path: /usr/bin/crun
    version: |-
      crun version 1.8.6
      commit: 73f759f4a39769f60990e7d225f561b4f4f06bcf
      rundir: /run/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +WASM:wasmedge +YAJL
  os: linux
  pasta:
    executable: /usr/bin/pasta
    package: passt-0^20230625.g32660ce-1.fc38.aarch64
    version: |
      pasta 0^20230625.g32660ce-1.fc38.aarch64
      Copyright Red Hat
      GNU Affero GPL version 3 or later <https://www.gnu.org/licenses/agpl-3.0.html>
      This is free software: you are free to change and redistribute it.
      There is NO WARRANTY, to the extent permitted by law.
  remoteSocket:
    exists: true
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-12.fc38.aarch64
    version: |-
      slirp4netns version 1.2.0
      commit: 656041d45cfca7a4176f6b7eed9e4fe6c11e8383
      libslirp: 4.7.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.3
  swapFree: 0
  swapTotal: 0
  uptime: 0h 1m 51.00s
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - docker.io
store:
  configFile: /usr/share/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mountopt: nodev,metacopy=on
  graphRoot: /var/lib/containers/storage
  graphRootAllocated: 106769133568
  graphRootUsed: 11724918784
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "true"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 44
  runRoot: /run/containers/storage
  transientStore: false
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 4.6.0
  Built: 1689942161
  BuiltTime: Fri Jul 21 21:22:41 2023
  GitCommit: ""
  GoVersion: go1.20.6
  Os: linux
  OsArch: linux/arm64
  Version: 4.6.0
```
```


### Podman in a container

No

### Privileged Or Rootless

Privileged

### Upstream Latest Release

Yes

### Additional environment details

Host OS: MacOS 13.5

### Additional information

_No response_

I'm trying to run prometheus on a mac with a mounted local volume:

```
podman machine init -v /Users/rrc:/Users/rrc
podman machine start
podman run -v "$PWD/promdata:/prometheus" prom/prometheus
```

Could this be related?

```
ts=2023-09-07T12:11:03.901Z caller=query_logger.go:93 level=error component=activeQueryTracker msg="Error opening query log file" file=/prometheus/queries.active err="open /prometheus/queries.active: permission denied"
panic: Unable to create mmap-ed active query log
```

`podman run -v $PWD/promdata:/prometheus ubuntu touch /prometheus/queries.active`  succeeds, so it seems I have normal write access to the file, but can't open it mmap'd.

--- a/pkg/machine/qemu/machine.go
+++ b/pkg/machine/qemu/machine.go
@@ -347,7 +347,7 @@ func (v *MachineVM) mountVolumesToVM(opts machine.StartOptions, name string) err
 		case MountType9p:
 			mountOptions := []string{"-t", "9p"}
 			mountOptions = append(mountOptions, []string{"-o", "trans=virtio", mount.Tag, mount.Target}...)
-			mountOptions = append(mountOptions, []string{"-o", "version=9p2000.L,msize=131072"}...)
+			mountOptions = append(mountOptions, []string{"-o", "version=9p2000.L,msize=131072,cache=mmap"}...)
 			if mount.ReadOnly {
 				mountOptions = append(mountOptions, []string{"-o", "ro"}...)
 			}

